sys/kern/vfs_aio.c

```diff
@@ -285,9 +285,13 @@ struct aiocb_ops {
 	int	(*store_status)(struct aiocb *ujob, long status);
 	int	(*store_error)(struct aiocb *ujob, long error);
 	int	(*store_kernelinfo)(struct aiocb *ujob, long jobref);
-	int	(*store_aiocb)(struct aiocb **ujobp, struct aiocb *ujob);
+	int	(*zero_aiocb)(struct aiocb **ujobp);
+	int	(*copyout_ujobptr)(const void *kaddr, void *uaddr, size_t len);
+	void	*(*memcpy_ujobptr)(void *dst, const void * src, size_t len);
```

Comment:
Do we really need the size?  The backend already knows the size (it's always '4' for compat32, and always 'sizeof(intcap_t)' in the CheriABI backend).  Ahh, this uses memcpy and copyout directly.  Hmm, I would still like to keep 'store_aiocb' as the op.  Even in CheriABI you are always calling copyoutcap, it's just a matter of what you are storing.  We could pass in the pointer to the in-kernel data structure to the callback so you can look inside of that, but I think preserving store_aiocb in that fashion will make for a smaller diff.

Reply:
I've restored something similar to store_aiocb and move the size into the operations.  Unfortunately, store_aiocb must have a different signature so every instance is changed.